packages/testing/src/chai-retry-plugin/helpers.ts

```diff
@@ -0,0 +1,48 @@
+import Chai from 'chai';
+import { timeout as addTimeoutToPromise, sleep } from 'promise-assist';
+
+import type { AssertionMethodStackItem, AssertionStackItem, RetryAndAssertArguments } from './types';
+
+const { expect } = Chai;
+
+function isAssertionMethodStackItem(item: AssertionStackItem): item is AssertionMethodStackItem {
+    return 'method' in item && 'args' in item;
+}
+
+const performRetries = async ({
+    functionToRetry,
+    options,
+    assertionStack,
+    isFunctionCallHandledByChai,
+}: RetryAndAssertArguments) => {
+    const { retries, delay } = options;
+    let retriesCount = 0;
+
+    while (retriesCount < retries) {
+        try {
+            retriesCount++;
+            const result = isFunctionCallHandledByChai ? functionToRetry : await functionToRetry();
+            let assertion = expect(result);
+
+            for (const assertionStackItem of assertionStack) {
```

Comment:
make
```
type AssertionStackItem = {
     propertyName: keyof Chai.Assertion;
     method?: AssertionMethod;
     args?: unknown[]
}
```

then you could replace this snippet with simpler and shorter one:
```
for (const { propertyName, method, args } of assertionStack) {
   if (method && args) {
       assertion = method.apply(assertion, args);
    } else {
        assertion = assertion[propertyName] as Chai.Assertion;
    }
}
```

Reply:
@yurii-ve fixed